content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+
+When compiling shaders with the "--use-spaces" argument, each SRG will take over a whole register space (or descriptor set, as known in Vulkan). AZSL abstracts the register and register space assignment of resource descriptors with the `ShaderResourceGroupSemantic` keyword, which must be specified when declaring SRGs.
+
+A Shader Resource Group is declared with the following syntax:
+  
+    ShaderResourceGroup <Name> : <Semantic>
+    {
+        <Data>
+    };
+     
+    //Alternatively
+    partial ShaderResourceGroup <Name> : <Semantic>
+    {
+        <Data>
+    };
+  
+* **&lt;Name&gt;** is the user defined name for the SRG.
+* **&lt;Semantic&gt;** is the name of the `ShaderResourceGroupSemantic` for the SRG. More on this later.
+* **&lt;Data&gt;** is one or more declaration of functions, constants and resources for the binding data contained in the SRG.  
+  
+In the code block shown above, a [partial](#partial) qualifier can be used when declaring SRGs.
+### &lt;Data&gt;
+SRGs can contain fundamental type variables like float, float3, int2, bool, matrices, etc. For each SRG, a Constant Buffer View (CBV) will be automatically created and it will pack all the fundamental type variables declared inside the SRG.
+
+SRGs can contain resource declarations like Texture2D, Texture2DMS, ByteAddressBuffer, etc. In general any HLSL shader resource of the categories:  Shader Resource Views (SRV), Unordered Access Views (UAV), Samplers and Constant Buffer Views (CBV).
+
+SRGs can contain unbounded arrays for resource declarations: SRVs, UAVs, CBVs and Samplers. There are some rules that need to be considered, depending on command line arguments when compiling with **AZSLc**. More on unbounded arrays [here](unbounded-arrays).
+
+SRGs can contain functions.
+
+SRGs can contain structs (declaration & instances). The Shader Build Pipeline and the runtime handle the platform-specific packing rules and alignment.
+
+SRGs can contain instances of classes (but you can not declare classes inside an SRG).  
+  
+**Example**  
+The following example demonstrates declaration of a Shader Resource Group. We have not covered details of what is a `ShaderResourceGroupSemantic`, but suffice to say, it defines the space (aka, descriptor set) index of the constants and resources declared inside the SRG.  
+  
+    ShaderResourceGroupSemantic BindingPerExample
+    {
+        FrequencyId = 0;
+    };
+       
+    ShaderResourceGroup ExampleSRG : BindingPerExample
+    {
+        float4          m_uniformColor;
+        Texture2D       m_texture;
+        Buffer<float>   m_buffer;
+        Sampler m_dynamicSampler;
+        Sampler m_staticSampler
+        {
+            AddressU = Wrap;
+            AddressV = Wrap;
+            AddressW = Wrap;
+            MagFilter = Linear;
+        };
+        
+        // You can define functions inside a ShaderResourceGroup
+        float GetRedColor()
+        {
+            return m_uniformColor.r;
+        }
+    };
+  
```

Comment:
- Use C++ code style in your code block by wrapping the code with three back ticks (```). (Remove the `\` when you make this change.)
- Remove indent. 
```suggestion
\```cpp  
ShaderResourceGroupSemantic BindingPerExample
{
  FrequencyId = 0;
};
 
ShaderResourceGroup ExampleSRG : BindingPerExample
{
  float4          m_uniformColor;
  Texture2D       m_texture;
  Buffer<float>   m_buffer;
  Sampler m_dynamicSampler;
  Sampler m_staticSampler
  {
      AddressU = Wrap;
      AddressV = Wrap;
      AddressW = Wrap;
      MagFilter = Linear;
  };
  
  // You can define functions inside a ShaderResourceGroup
  float GetRedColor()
  {
      return m_uniformColor.r;
  }
  };
\```  
```

Reply:
Fixed.